README.md

```diff
@@ -35,25 +35,354 @@ After signing up for a Moesif account, your Moesif Application Id will be displa
 6. Go back to [Cloudflare Workers Dashboard](https://dash.cloudflare.com/workers) and select `Add Route`.
 * `Route` : Set a valid route. We recommend matching all requests for your domain. For example if your domain is `acmeinc.com`, your pattern should be `*acmeinc.com/*`.
 * `Worker`: Select the worker created above.
-* Save. 
+* Save.
 
 #### Try it!
 Just visit your link on cloudflare using your web browser `https://my-cloudflare-domain/my-path` or
 ```bash
 curl https://my-cloudflare-domain/my-path
 ```
-The API calls should show up in Moesif event stream. 
+The API calls should show up in Moesif event stream.
 
 ## [Alternative] Install via Cloudflare Workers (Custom install) using `wrangler` CLI
-Using `Cloudflare/Wrangler` CLI allows for automated install using `wrangler publish`, as well as view/`tail` using `wrangler tail` command 
-* Visit [WRANGLER.md](WRANGLER.md) for details.
+Using `Cloudflare/Wrangler` CLI allows for automated install using `wrangler publish`, as well as view/`tail` using `wrangler tail` command
+* Visit [WRANGLER.md](https://github.com/Moesif/moesif-cloudflare/blob/master/WRANGLER.md) for details.
 
 
 ## Advanced Usage
-* Visit [AdvancedUsage.md](AdvancedUsage.md)
+
+### Optional: Advanced usage
+
+1. Custom routes and using multiple Moesif Application Ids
+
+Here is a sample `INSTALL_OPTIONS.urlPatterns`:
+```javascript
+urlPatterns = [
+  {"applicationId" : "<Moesif AppId 1>", "regex": "^https://stating.my.domain/hello"},
+  {"applicationId" : "<Moesif AppId 2>", "regex": "^https://experiment.my.domain/hello"},
+  {"applicationId" : "",                 "regex": "^https://www.my.domain/hello"}
+]
+```
+A blank `applicationId` in `urlPatterns` will result in use of default `INSTALL_OPTIONS.applicationId`
+
+2. Customizing `addEventListener` to work with other unrelated 3rd party `Cloudflare Apps` or other advanced worker scenarios
+Advanced scenarios are not discussed here. However, when working with other apps or workers, you may need to replace `event.respondWith(logRequest(event));` with `event.waitUntil(logRequest(event));`
+
+3. Additional options are below:
+
+```javascript
+INSTALL_OPTIONS = {
+    // Your Moesif Application Id
+    "applicationId": "Your Moesif Application Id",
+
+    // Only used by CloudFlare App Worker Framework. Modify identifyUser() function instead.
+    "userIdHeader": "",
+
+    // Only used by CloudFlare App Worker Framework. Modify identifyCompany() function instead.
+    "companyIdHeader": "",
+
+    // Only used by CloudFlare App Worker Framework. Modify getSessionToken() function instead.
+    "sessionTokenHeader": "",
+
+    // true or false
+    "hideCreditCards": true,
+
+    // set to true to prevent insertion of X-Moesif-Transaction-Id
+    "disableTransactionId": false,
+
+    // Log incoming API calls hitting your Cloudflare Worker
+    "logIncomingRequests": true,
+
+    // Log outgoing calls to your origin server from your Cloudflare Worker
```

Comment:
Should this match what we write in install.json `"Log responses from your origin server."`?

Reply:
ok.

Reply:
I will clarify in comments. unfortunately this is a huge misnomer added a while back. That results in confusion why the response isn't logged for tallyfy